### PR TITLE
GitHub-issue#253 : Implemented GeoIP processor multiple IP's function…

### PR DESCRIPTION
**Pull request by venkataraopasyavula**
### Description
Implementation of GeoIP processor multiple IP's support
 
### Issues Resolved
GitHub-Issue #253 
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by kkondaka — APPROVED** [review]

File: data-prepper-plugins/geoip-processor/README.md [hunk 1]
```diff
@@ -46,9 +46,9 @@ pipeline:
 
 - `keys` (Required) : List of properties like source, target and attributes can be specified where the location fields are written
 
-  - `source` (Required) : source IP for which enrichment will be done. Public IP can be either IPV4 or IPV6.
+  - `source` (Required) : source  and destination IP's for which enrichment will be done. Public IP's can be either IPV4 or IPV6.
```
Comment (JonahCalvo) [inline data-prepper-plugins/geoip-processor/README.md:49]: Im a little confused, what does destination IP refer to? Does this just mean if there are two IPs that want to be looked up?

File: data-prepper-plugins/geoip-processor/README.md [hunk 2]
```diff
@@ -68,27 +68,11 @@ pipeline:
...
-"location" : {
-"status" : "success"
-"country" : "United States"
-"city_name" : "Seattle"
-"latitude" : "47.64097"
-"longitude" : "122.25894"
-"zip_code" : "98115"
-}
-
+{"peer":{"ips":{"src_ip":"8.8.8.8","dst_ip":"8.8.8.9"},"host":"example.org"},"status":"success","target1":{"continent_name":"North America","country_iso_code":"US","timezone":"America/Chicago","ip":"8.8.8.8","country_name":"United States","location":{"lon":-97.822,"lat":37.751},"organization_name":"GOOGLE","asn":15169,"network":"8.8.8.0/24"},"target2":{"continent_name":"North America","country_iso_code":"US","timezone":"America/Chicago","ip":"8.8.8.9","country_name":"United States","location":{"lon":-97.822,"lat":37.751},"organization_name":"GOOGLE","asn":15169,"network":"8.8.8.0/24"}}
```
Comment (JonahCalvo) [inline data-prepper-plugins/geoip-processor/README.md:75]: Seems like formatting got lost on this line

File: data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorUrlServiceIT.java [hunk 3]
```diff
@@ -42,9 +41,9 @@ public class GeoIPProcessorUrlServiceIT {
     @BeforeEach
     public void setUp() throws JsonProcessingException {
 
-        maxmindLicenseKey = System.getProperty("tests.geoipProcessor.maxmindLicenseKey");
+        maxmindLicenseKey =  System.getProperty("tests.geoipProcessor.maxmindLicenseKey");
```
Comment (JonahCalvo) [inline data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorUrlServiceIT.java:44]: nit: whitespace

File: data-prepper-plugins/geoip-processor/README.md (hunk 1 above)
Comment (venkataraopasyavula) [reply data-prepper-plugins/geoip-processor/README.md:49]: Yes. There can be multiple IP's with  "," is the delimiter with any name. However the same name should be present in YAML cinfiguration for parsing and reading the IP. For example as below:
source: ["/peer/ips/src_ip", "/peer/ips/dest_ip"]

File: data-prepper-plugins/geoip-processor/README.md (hunk 2 above)
Comment (venkataraopasyavula) [reply data-prepper-plugins/geoip-processor/README.md]: Formatted now with proper JSON formatter. Please check now.

File: data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorUrlServiceIT.java (hunk 3 above)
Comment (venkataraopasyavula) [reply data-prepper-plugins/geoip-processor/src/integrationTest/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorUrlServiceIT.java]: Removed whitespace. Please check now.

**Review by dlvenable — CHANGES_REQUESTED** [review]
Thank you for this PR @venkataraopasyavula .

I'm not quite sure I understand why you are making `source` and `target` into lists. This plugin already supports multiple `key` values. So each `key` relates a `source` to a `target`.

Why add a second list therein? Can you explain this?

Thanks.

**Comment by dlvenable** [comment]
We are closing this task as obsolete.

**Pull request closed without merging**